testing/main.lua

```diff
@@ -0,0 +1,216 @@
+        module = arglist[a] 
+        table.insert(modules, module)
+      elseif module ~= '' and love[module] ~= nil and method == '' then
+        if love.test[module][arglist[a]] ~= nil then method = arglist[a] end
+      end
+    end
+    if testcmd == '--runSpecificModules' then
+      if love[ arglist[a] ] ~= nil and arglist[a] ~= '--isRunner' then 
+        table.insert(modules, arglist[a]) 
+      end
+    end
+    if arglist[a] == '--runSpecificMethod' then
+      testcmd = arglist[a]
+      modules = {}
+    end
+    if arglist[a] == '--runSpecificModules' then
+      testcmd = arglist[a]
+      modules = {}
+    end
+    if arglist[a] == '--isRunner' then
+      GITHUB_RUNNER = true
+    end
+  end
+
+  -- runSpecificMethod uses the module + method given
+  if testcmd == '--runSpecificMethod' then
+    local testmodule = TestModule:new(module, method)
+    table.insert(love.test.modules, testmodule)
+    if module ~= '' and method ~= '' then
+      love.test.module = testmodule
+      love.test.module:log('grey', '--runSpecificMethod "' .. module .. '" "' .. method .. '"')
+      love.test.output = 'lovetest_runSpecificMethod_' .. module .. '_' .. method
+    else
+      if method == '' then cmderr = 'No valid method specified' end
+      if module == '' then cmderr = 'No valid module specified' end
+    end
+  end
+
+  -- runSpecificModules runs all methods for all the modules given
+  if testcmd == '--runSpecificModules' then
+    local modulelist = {}
+    for m=1,#modules do
+      local testmodule = TestModule:new(modules[m])
+      table.insert(love.test.modules, testmodule)
+      table.insert(modulelist, modules[m])
+    end
+    if #modulelist > 0 then
+      love.test.module = love.test.modules[1]
+      love.test.module:log('grey', '--runSpecificModules "' .. table.concat(modulelist, '" "') .. '"')
+      love.test.output = 'lovetest_runSpecificModules_' .. table.concat(modulelist, '_')
+    else
+      cmderr = 'No modules specified'
+    end
+  end
+
+  -- otherwise default runs all methods for all modules
+  if arglist[1] == nil or arglist[1] == '' or arglist[1] == '--runAllTests' then
+    for m=1,#modules do
+      local testmodule = TestModule:new(modules[m])
+      table.insert(love.test.modules, testmodule)
+    end
+    love.test.module = love.test.modules[1]
+    love.test.module:log('grey', '--runAllTests')
+    love.test.output = 'lovetest_runAllTests'
+  end
+
+  if GITHUB_RUNNER then
+    love.test.module:log('grey', '--isRunner')
+  end
+
+  -- invalid command
+  if love.test.module == nil then
+    print(cmderr)
+    love.event.quit(0)
+  else 
+    -- start first module
+    TextCommand:set(testcmd)
+    love.test.module:runTests()
+  end
+
+end
+
+-- love.update
+-- run test suite logic 
+love.update = function(delta)
+  love.test:runSuite(delta)
+end
+
+
+-- love.draw
+-- draw a little logo to the screen
+love.draw = function()
+  local lw = (love.graphics.getPixelWidth() - 128) / 2

```

Comment:
I think this will draw off-screen when high dpi is used - `love.graphics.getWidth` (and getHeight for the line below) should be used instead